webapp/components/file_upload.jsx

```diff
@@ -72,6 +72,8 @@ class FileUpload extends React.Component {
     }
 
     uploadFiles(files) {
+        const sortedFiles = Array.from(files).sort((a, b) => a.name.localeCompare(b.name));
```

Comment:
I think you need to get the locale from the store. Take a look at https://github.com/mattermost/platform/blob/master/webapp/utils/channel_utils.jsx#L100 where we sort the channels on the sidebar.